### ISSUE TITLE
Undeclared error on xx$k=xx$c/xx$t if $t not existing and index time is repeated

### ISSUE THREAD

**Issue by Rloveeee** [joshuaulrich/xts#265]
### Description

Undeclared error on `xx$k=xx$c/xx$t` if `$t` not existing and index time is repeated.
And `$k` is given the value of the last column which disappeared.

### Expected behavior

Should declared an error as it's done in zoo and in xts if time is not repeated.
I am using a lot of xts with repeated time and so far it has been working pretty well.

### Minimal, reproducible example

```r
xx=xts(data.frame("c"=1:5,"c1"=2:6),Sys.Date()+c(1,1,1,2,2))
xx$k=xx$c/xx$t
### Output
print(xx)
#            c k
# 2018-08-24 1 2
# 2018-08-24 2 3
# 2018-08-24 3 4
```

Column `c1` has changed name to`k`!

### Session Info

```r
R version 3.5.0 (2018-04-23)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 17134)

Matrix products: default

locale:
[1] LC_COLLATE=French_France.1252  LC_CTYPE=French_France.1252   
[3] LC_MONETARY=French_France.1252 LC_NUMERIC=C                  
[5] LC_TIME=French_France.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] xts_0.11-0 zoo_1.8-3 

loaded via a namespace (and not attached):
[1] compiler_3.5.0  grid_3.5.0      lattice_0.20-35
```

Thanks for this marvelous package that I use everyday.

**Comment by joshuaulrich** [comment]
Thanks for the report!  Can you please edit your session info to include all the output from `sessionInfo()`?

You say "should declare an error as it's done in zoo".  Can you give an example of when this throws an error for zoo objects?

**Comment by joshuaulrich** [comment]
Here is the what throws an error in zoo:

```r
zz <- zoo(data.frame("c"=1:5,"c1"=2:6),Sys.Date()+c(1,1,1,2,2))
# Warning message:
# In zoo(data.frame(c = 1:5, c1 = 2:6), Sys.Date() + c(1, 1, 1, 2,  :
#   some methods for "zoo" objects do not work if the index entries in 'order.by' are not unique
zz$k <- zz$c/zz$t
# Error in merge.zoo(..., all = all, fill = fill, suffixes = suffixes, retclass = "zoo",  : 
#   series cannot be merged with non-unique index entries in a series
```

**Comment by joshuaulrich** [comment]
This is related to #45.  Thanks to Achim, it is fixed in zoo r1144, version 1.8-6.

```r
R> xx=xts(data.frame("c"=1:5,"c1"=2:6),Sys.Date()+c(1,1,1,2,2))
R> xx$k=xx$c/xx$t
R> ### Output
R> print(xx)
           c c1
2019-04-17 1  2
2019-04-17 2  3
2019-04-17 3  4
2019-04-18 4  5
2019-04-18 5  6
R> packageVersion("zoo")
[1] '1.8.6'
```
